### PR TITLE
[#591] Update dotnet bots to target net 6 - Add .NETCore3.1 bots

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses # 591

## Description
This PR adds the SimpleHostBotDotNet-3.1 and the EchoSkillBotDotNet-3.1 bots to the project.

### Detailed Changes
- Added two new projects to the `Microsoft.Bot.Builder.FunctionalTestsBots` solution:
   - Microsoft.Bot.Builder.FunctionalTestsBots.EchoSkillBot31
   - Microsoft.Bot.Builder.FunctionalTestsBots.SimpleHostBot31

## Testing
The following images show the new bots added working as expected.
image
image

### CODE REVIEW

File: Bots/DotNet/EchoSkillBot-3.1/Microsoft.Bot.Builder.FunctionalTestsBots.EchoSkillBot31.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <LangVersion>latest</LangVersion>
+    <AssemblyName>Microsoft.BotFrameworkFunctionalTests.EchoSkillBot</AssemblyName>
+    <RootNamespace>Microsoft.BotFrameworkFunctionalTests.EchoSkillBot</RootNamespace>
```
Comment (sw-joelmut) [inline Bots/DotNet/EchoSkillBot-3.1/Microsoft.Bot.Builder.FunctionalTestsBots.EchoSkillBot31.csproj]: Should we change it to 31 here as well?, like dotnet 2.1 at the time?
image

File: Bots/DotNet/EchoSkillBot-3.1/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "$schema": "http://json.schemastore.org/launchsettings.json",
+  "iisSettings": {
+    "windowsAuthentication": false,
+    "anonymousAuthentication": true,
+    "iisExpress": {
+      "applicationUrl": "http://localhost:35405/",
```
Comment (sw-joelmut) [inline Bots/DotNet/EchoSkillBot-3.1/Properties/launchSettings.json:7]: Dont remember the full list of ports, just mentioning this to ensure is in the correct one

**Review by sw-joelmut — CHANGES_REQUESTED** [review]
Same cases for host

File: Bots/DotNet/EchoSkillBot-3.1/Properties/launchSettings.json (hunk 2 above)
Comment (ceciliaavila) [reply Bots/DotNet/EchoSkillBot-3.1/Properties/launchSettings.json:7]: I used the ports that the old netcore 2.1 bots were using.

File: Bots/DotNet/EchoSkillBot-3.1/Microsoft.Bot.Builder.FunctionalTestsBots.EchoSkillBot31.csproj (hunk 1 above)
Comment (ceciliaavila) [reply Bots/DotNet/EchoSkillBot-3.1/Microsoft.Bot.Builder.FunctionalTestsBots.EchoSkillBot31.csproj]: Updated. Thanks!

**Review by sw-joelmut — APPROVED** [review]

**Pull request closed without merging**